libkbfs/keybase_service_base.go

```diff
@@ -619,3 +619,25 @@ func (k *KeybaseServiceBase) GetTLFCryptKeys(ctx context.Context,
 
 	return res, nil
 }
+
+// GetTLFCanonicalID implements the TlfKeysInterface interface for
+// KeybaseServiceBase.
+func (k *KeybaseServiceBase) GetTLFCanonicalID(ctx context.Context,
+	arg keybase1.GetTLFCanonicalIDArg) (res keybase1.TLFCanonicalID, err error) {
+	ctx = ctxWithRandomIDReplayable(ctx, CtxKeybaseServiceIDKey, CtxKeybaseServiceOpID,
+		k.log)
+	tlfHandle, err := k.getHandleFromFolderName(ctx, arg.TlfName, arg.Public)
```

Comment:
Can we return an error if `!arg.Public`?  I really want to discourage direct use of the TLF ID whenever possible, so ideally we'd only allow this used in the strictest possible way. (I realize `GetTLFCryptKeys` also returns the ID, which also makes me uncomfortable. but I think that's slightly better since the name implies it should only be used for something specific.)